data-prepper-plugins/opensearch/src/main/java/com/amazon/dataprepper/plugins/sink/opensearch/index/IndexType.java

```diff
@@ -1,7 +1,42 @@
 package com.amazon.dataprepper.plugins.sink.opensearch.index;
 
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
 public enum IndexType {
-    TRACE_ANALYTICS_RAW,
-    TRACE_ANALYTICS_SERVICE_MAP,
-    CUSTOM;
+    TRACE_ANALYTICS_RAW("trace-analytics-raw"),
+    TRACE_ANALYTICS_SERVICE_MAP("trace-analytics-service-map"),
+    CUSTOM("custom");
+
+    private final String valueString;
+
+    // Reverse-lookup map for getting a IndexType from a name
+    private static final Map<String, IndexType> STRING_TO_INDEX_TYPE_MAP = new HashMap<>();
+
+    static {
+        Arrays.stream(IndexType.values())
+                .forEach(indexType -> STRING_TO_INDEX_TYPE_MAP.put(indexType.valueString, indexType));
+    }
+
+    IndexType(final String name){
+        this.valueString = name;
+    }
```

Comment:
I would recommend keeping your naming consistent throughout. Right now you are referring to the same value as a `valueString` and a `name`. Pick either `value` or `name`. 

Then your `get` function can be renamed to `getByValue` or `getByName` and this should address @dlvenable concern as well.

Reply:
My mistake. I considered several options.  "name" does not work. Java enums consider the primary enum string, e.g. TRACE_ANALYTICS_RAW in our case, as "name"
I am changing all of them to value.